lib/record.rb

```diff
@@ -0,0 +1,23 @@
+class Record
+  attr_reader :amount, :date, :type
+
+  def initialize(amount, date)
+    @amount = amount
+    @date = date
+    @type = set_type
+  end
+
+  def add_to(ledger)
+    ledger.transactions << format_record
```

Comment:
Is there a downside to formatting the record when its added to the ledger, rather than when it needs to be printed?